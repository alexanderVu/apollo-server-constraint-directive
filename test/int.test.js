const { createTestClient } = require('apollo-server-testing');
const { gql } = require('apollo-server');
const { constructTestServer } = require('./__setup');
const SET_BOOK = gql`mutation createBook($input: BookInput) {
  createBook(input: $input) {
    title
  }
}`
const GET_BOOK = gql`query {
  books {
    title
  }
}`
const resolvers = function (data) {
  return {
    Query: {
      books () {
        return data
      }
    }
  }
}
// INPUT_FIELD_DEFINITION
describe('INPUT_FIELD_DEFINITION Int validate @constraint #min', () => {
  const typeDefs = gql`
    directive @constraint(min: Int) on INPUT_FIELD_DEFINITION
    type Query {
      books: [Book]
    }
    type Book {
      title: Int
    }
    type Mutation {
      createBook(input: BookInput): Book
    }
    input BookInput {
      title: Int! @constraint(min: 3)
    }
  `;

  it('should pass', async () => {
    const { server } = constructTestServer({ typeDefs });
    const { query } = createTestClient(server);
    const res = await query({ mutation: SET_BOOK, variables: { input: { title: 3 } } });
    expect(res).toMatchSnapshot();
  });

  it('should fail', async () => {
    const { server } = constructTestServer({ typeDefs });
    const { query } = createTestClient(server);
    const res = await query({ mutation: SET_BOOK, variables: { input: { title: 2 } } });
    expect(res).toMatchSnapshot();
  });
});

describe('INPUT_FIELD_DEFINITION Int validate @constraint #max', () => {
  const typeDefs = gql`
    directive @constraint(max: Int) on INPUT_FIELD_DEFINITION
    type Query {
      books: [Book]
    }
    type Book {
      title: Int
    }
    type Mutation {
      createBook(input: BookInput): Book
    }
    input BookInput {
      title: Int! @constraint(max: 3)
    }
  `;

  it('should pass', async () => {
    const { server } = constructTestServer({ typeDefs });
    const { query } = createTestClient(server);
    const res = await query({ mutation: SET_BOOK, variables: { input: { title: 2 } } });
    expect(res).toMatchSnapshot();
  });

  it('should fail', async () => {
    const { server } = constructTestServer({ typeDefs });
    const { query } = createTestClient(server);
    const res = await query({ mutation: SET_BOOK, variables: { input: { title: 5 } } });
    expect(res).toMatchSnapshot();
  });
});

describe('INPUT_FIELD_DEFINITION Int validate @constraint #exclusiveMin', () => {
  const typeDefs = gql`
    directive @constraint(exclusiveMin: Int) on INPUT_FIELD_DEFINITION
    type Query {
      books: [Book]
    }
    type Book {
      title: Int
    }
    type Mutation {
      createBook(input: BookInput): Book
    }
    input BookInput {
      title: Int! @constraint(exclusiveMin: 3)
    }
  `;

  it('should pass', async () => {
    const { server } = constructTestServer({ typeDefs });
    const { query } = createTestClient(server);
    const res = await query({ mutation: SET_BOOK, variables: { input: { title: 4 } } });
    expect(res).toMatchSnapshot();
  });

  it('should fail', async () => {
    const { server } = constructTestServer({ typeDefs });
    const { query } = createTestClient(server);
    const res = await query({ mutation: SET_BOOK, variables: { input: { title: 3 } } });
    expect(res).toMatchSnapshot();
  });
});

describe('INPUT_FIELD_DEFINITION Int validate @constraint #exclusiveMax', () => {
  const typeDefs = gql`
    directive @constraint(exclusiveMax: Int) on INPUT_FIELD_DEFINITION
    type Query {
      books: [Book]
    }
    type Book {
      title: Int
    }
    type Mutation {
      createBook(input: BookInput): Book
    }
    input BookInput {
      title: Int! @constraint(exclusiveMax: 3)
    }
  `;

  it('should pass', async () => {
    const { server } = constructTestServer({ typeDefs });
    const { query } = createTestClient(server);
    const res = await query({ mutation: SET_BOOK, variables: { input: { title: 2 } } });
    expect(res).toMatchSnapshot();
  });

  it('should fail', async () => {
    const { server } = constructTestServer({ typeDefs });
    const { query } = createTestClient(server);
    const res = await query({ mutation: SET_BOOK, variables: { input: { title: 3 } } });
    expect(res).toMatchSnapshot();
  });
});

describe('INPUT_FIELD_DEFINITION Int validate @constraint #multipleOf', () => {
  const typeDefs = gql`
    directive @constraint(multipleOf: Int) on INPUT_FIELD_DEFINITION
    type Query {
      books: [Book]
    }
    type Book {
      title: Int
    }
    type Mutation {
      createBook(input: BookInput): Book
    }
    input BookInput {
      title: Int! @constraint(multipleOf: 2)
    }
  `;

  it('should pass', async () => {
    const { server } = constructTestServer({ typeDefs });
    const { query } = createTestClient(server);
    const res = await query({ mutation: SET_BOOK, variables: { input: { title: 10 } } });
    expect(res).toMatchSnapshot();
  });

  it('should fail', async () => {
    const { server } = constructTestServer({ typeDefs });
    const { query } = createTestClient(server);
    const res = await query({ mutation: SET_BOOK, variables: { input: { title: 7 } } });
    expect(res).toMatchSnapshot();
  });
});

// FIELD_DEFINITION
describe('FIELD_DEFINITION Int validate @constraint #min', () => {
  const typeDefs = gql`
    directive @constraint(min: Int) on FIELD_DEFINITION
    type Query {
      books: [Book]
    }
    type Book {
      title: Int @constraint(min: 3)
    }
  `;

  it('should pass', async () => {
    const mockData = [{ title: 2 }, { title: 3 }]
    const { server } = constructTestServer({ typeDefs, resolvers: resolvers(mockData) })
    const { query } = createTestClient(server)
    const res = await query({ query: GET_BOOK })
    expect(res).toMatchSnapshot()
  });

  it('should fail', async () => {
    const mockData = [{ title: 1 }, { title: 2 }]
    const { server } = constructTestServer({ typeDefs, resolvers: resolvers(mockData) })
    const { query } = createTestClient(server)
    const res = await query({ query: GET_BOOK })
    expect(res).toMatchSnapshot()
  });
});

describe('FIELD_DEFINITION Int validate @constraint #min0', () => {
  const typeDefs = gql`
    directive @constraint(min: Int) on FIELD_DEFINITION
    type Query {
      books: [Book]
    }
    type Book {
      title: Int @constraint(min: 0)
    }
  `;

  it('should pass', async () => {
    const mockData = [{ title: 1 }, { title: 2 }]
    const { server } = constructTestServer({ typeDefs, resolvers: resolvers(mockData) })
    const { query } = createTestClient(server)
    const res = await query({ query: GET_BOOK })
    expect(res).toMatchSnapshot()
  });

  it('should fail', async () => {
    const mockData = [{ title: -1 }, { title: 2 }]
    const { server } = constructTestServer({ typeDefs, resolvers: resolvers(mockData) })
    const { query } = createTestClient(server)
    const res = await query({ query: GET_BOOK })
    expect(res).toMatchSnapshot()
  });
});

describe('FIELD_DEFINITION Int validate @constraint #max', () => {
  const typeDefs = gql`
    directive @constraint(max: Int) on FIELD_DEFINITION
    type Query {
      books: [Book]
    }
    type Book {
      title: Int @constraint(max: 2)
    }
  `;

  it('should pass', async () => {
    const mockData = [{ title: 1 }, { title: 2 }]
    const { server } = constructTestServer({ typeDefs, resolvers: resolvers(mockData) })
    const { query } = createTestClient(server)
    const res = await query({ query: GET_BOOK })
    expect(res).toMatchSnapshot()
  });

  it('should fail', async () => {
    const mockData = [{ title: 3 }, { title: 2 }]
    const { server } = constructTestServer({ typeDefs, resolvers: resolvers(mockData) })
    const { query } = createTestClient(server)
    const res = await query({ query: GET_BOOK })
    expect(res).toMatchSnapshot()
  });
});

describe('FIELD_DEFINITION Int validate @constraint #max0', () => {
  const typeDefs = gql`
    directive @constraint(max: Int) on FIELD_DEFINITION
    type Query {
      books: [Book]
    }
    type Book {
      title: Int @constraint(max: 0)
    }
  `;

  it('should pass', async () => {
    const mockData = [{ title: 0 }, { title: -1 }]
    const { server } = constructTestServer({ typeDefs, resolvers: resolvers(mockData) })
    const { query } = createTestClient(server)
    const res = await query({ query: GET_BOOK })
    expect(res).toMatchSnapshot()
  });

  it('should fail', async () => {
    const mockData = [{ title: 3 }, { title: -2 }]
    const { server } = constructTestServer({ typeDefs, resolvers: resolvers(mockData) })
    const { query } = createTestClient(server)
    const res = await query({ query: GET_BOOK })
    expect(res).toMatchSnapshot()
  });
});

describe('FIELD_DEFINITION Int validate @constraint #exclusiveMin', () => {
  const typeDefs = gql`
    directive @constraint(exclusiveMin: Int) on FIELD_DEFINITION
    type Query {
      books: [Book]
    }
    type Book {
      title: Int @constraint(exclusiveMin: 2)
    }
  `;

  it('should pass', async () => {
    const mockData = [{ title: 3 }, { title: 4 }]
    const { server } = constructTestServer({ typeDefs, resolvers: resolvers(mockData) })
    const { query } = createTestClient(server)
    const res = await query({ query: GET_BOOK })
    expect(res).toMatchSnapshot()
  });

  it('should fail', async () => {
    const mockData = [{ title: 3 }, { title: 2 }]
    const { server } = constructTestServer({ typeDefs, resolvers: resolvers(mockData) })
    const { query } = createTestClient(server)
    const res = await query({ query: GET_BOOK })
    expect(res).toMatchSnapshot()
  });
});

describe('FIELD_DEFINITION Int validate @constraint #exclusiveMin0', () => {
  const typeDefs = gql`
    directive @constraint(exclusiveMin: Int) on FIELD_DEFINITION
    type Query {
      books: [Book]
    }
    type Book {
      title: Int @constraint(exclusiveMin: 0)
    }
  `;

  it('should pass', async () => {
    const mockData = [{ title: 1 }, { title: 3 }]
    const { server } = constructTestServer({ typeDefs, resolvers: resolvers(mockData) })
    const { query } = createTestClient(server)
    const res = await query({ query: GET_BOOK })
    expect(res).toMatchSnapshot()
  });

  it('should fail', async () => {
    const mockData = [{ title: 0 }, { title: 2 }]
    const { server } = constructTestServer({ typeDefs, resolvers: resolvers(mockData) })
    const { query } = createTestClient(server)
    const res = await query({ query: GET_BOOK })
    expect(res).toMatchSnapshot()
  });
});

describe('FIELD_DEFINITION Int validate @constraint #exclusiveMax', () => {
  const typeDefs = gql`
    directive @constraint(exclusiveMax: Int) on FIELD_DEFINITION
    type Query {
      books: [Book]
    }
    type Book {
      title: Int @constraint(exclusiveMax: 2)
    }
  `;

  it('should pass', async () => {
    const mockData = [{ title: 0 }, { title: 1 }]
    const { server } = constructTestServer({ typeDefs, resolvers: resolvers(mockData) })
    const { query } = createTestClient(server)
    const res = await query({ query: GET_BOOK })
    expect(res).toMatchSnapshot()
  });

  it('should fail', async () => {
    const mockData = [{ title: 1 }, { title: 2 }]
    const { server } = constructTestServer({ typeDefs, resolvers: resolvers(mockData) })
    const { query } = createTestClient(server)
    const res = await query({ query: GET_BOOK })
    expect(res).toMatchSnapshot()
  });
});

describe('FIELD_DEFINITION Int validate @constraint #exclusiveMax0', () => {
  const typeDefs = gql`
    directive @constraint(exclusiveMax: Int) on FIELD_DEFINITION
    type Query {
      books: [Book]
    }
    type Book {
      title: Int @constraint(exclusiveMax: 0)
    }
  `;

  it('should pass', async () => {
    const mockData = [{ title: -1 }, { title: -2 }]
    const { server } = constructTestServer({ typeDefs, resolvers: resolvers(mockData) })
    const { query } = createTestClient(server)
    const res = await query({ query: GET_BOOK })
    expect(res).toMatchSnapshot()
  });

  it('should fail', async () => {
    const mockData = [{ title: 0 }, { title: -2 }]
    const { server } = constructTestServer({ typeDefs, resolvers: resolvers(mockData) })
    const { query } = createTestClient(server)
    const res = await query({ query: GET_BOOK })
    expect(res).toMatchSnapshot()
  });
});

describe('FIELD_DEFINITION Int validate @constraint #multipleOf', () => {
  const typeDefs = gql`
    directive @constraint(multipleOf: Int) on FIELD_DEFINITION
    type Query {
      books: [Book]
    }
    type Book {
      title: Int @constraint(multipleOf: 2)
    }
  `;

  it('should pass', async () => {
    const mockData = [{ title: 2 }, { title: 4 }, { title: 6 }]
    const { server } = constructTestServer({ typeDefs, resolvers: resolvers(mockData) })
    const { query } = createTestClient(server)
    const res = await query({ query: GET_BOOK })
    expect(res).toMatchSnapshot()
  });

  it('should fail', async () => {
    const mockData = [{ title: 1 }, { title: 2 }, { title: 3 }]
    const { server } = constructTestServer({ typeDefs, resolvers: resolvers(mockData) })
    const { query } = createTestClient(server)
    const res = await query({ query: GET_BOOK })
    expect(res).toMatchSnapshot()
  });
});
