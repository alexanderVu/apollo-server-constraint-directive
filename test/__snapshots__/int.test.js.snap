// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FIELD_DEFINITION Int validate @constraint #exclusiveMax should fail 1`] = `
Object {
  "data": Object {
    "books": Array [
      Object {
        "title": 1,
      },
      Object {
        "title": null,
      },
    ],
  },
  "errors": Array [
    [GraphQLError: Must be less than 2],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`FIELD_DEFINITION Int validate @constraint #exclusiveMax should pass 1`] = `
Object {
  "data": Object {
    "books": Array [
      Object {
        "title": 0,
      },
      Object {
        "title": 1,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`FIELD_DEFINITION Int validate @constraint #exclusiveMax0 should fail 1`] = `
Object {
  "data": Object {
    "books": Array [
      Object {
        "title": null,
      },
      Object {
        "title": -2,
      },
    ],
  },
  "errors": Array [
    [GraphQLError: Must be less than 0],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`FIELD_DEFINITION Int validate @constraint #exclusiveMax0 should pass 1`] = `
Object {
  "data": Object {
    "books": Array [
      Object {
        "title": -1,
      },
      Object {
        "title": -2,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`FIELD_DEFINITION Int validate @constraint #exclusiveMin should fail 1`] = `
Object {
  "data": Object {
    "books": Array [
      Object {
        "title": 3,
      },
      Object {
        "title": null,
      },
    ],
  },
  "errors": Array [
    [GraphQLError: Must be greater than 2],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`FIELD_DEFINITION Int validate @constraint #exclusiveMin should pass 1`] = `
Object {
  "data": Object {
    "books": Array [
      Object {
        "title": 3,
      },
      Object {
        "title": 4,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`FIELD_DEFINITION Int validate @constraint #exclusiveMin0 should fail 1`] = `
Object {
  "data": Object {
    "books": Array [
      Object {
        "title": null,
      },
      Object {
        "title": 2,
      },
    ],
  },
  "errors": Array [
    [GraphQLError: Must be greater than 0],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`FIELD_DEFINITION Int validate @constraint #exclusiveMin0 should pass 1`] = `
Object {
  "data": Object {
    "books": Array [
      Object {
        "title": 1,
      },
      Object {
        "title": 3,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`FIELD_DEFINITION Int validate @constraint #max should fail 1`] = `
Object {
  "data": Object {
    "books": Array [
      Object {
        "title": null,
      },
      Object {
        "title": 2,
      },
    ],
  },
  "errors": Array [
    [GraphQLError: Must be no greater than 2],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`FIELD_DEFINITION Int validate @constraint #max should pass 1`] = `
Object {
  "data": Object {
    "books": Array [
      Object {
        "title": 1,
      },
      Object {
        "title": 2,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`FIELD_DEFINITION Int validate @constraint #max0 should fail 1`] = `
Object {
  "data": Object {
    "books": Array [
      Object {
        "title": null,
      },
      Object {
        "title": -2,
      },
    ],
  },
  "errors": Array [
    [GraphQLError: Must be no greater than 0],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`FIELD_DEFINITION Int validate @constraint #max0 should pass 1`] = `
Object {
  "data": Object {
    "books": Array [
      Object {
        "title": 0,
      },
      Object {
        "title": -1,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`FIELD_DEFINITION Int validate @constraint #min should fail 1`] = `
Object {
  "data": Object {
    "books": Array [
      Object {
        "title": null,
      },
      Object {
        "title": null,
      },
    ],
  },
  "errors": Array [
    [GraphQLError: Must be at least 3],
    [GraphQLError: Must be at least 3],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`FIELD_DEFINITION Int validate @constraint #min should pass 1`] = `
Object {
  "data": Object {
    "books": Array [
      Object {
        "title": null,
      },
      Object {
        "title": 3,
      },
    ],
  },
  "errors": Array [
    [GraphQLError: Must be at least 3],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`FIELD_DEFINITION Int validate @constraint #min0 should fail 1`] = `
Object {
  "data": Object {
    "books": Array [
      Object {
        "title": null,
      },
      Object {
        "title": 2,
      },
    ],
  },
  "errors": Array [
    [GraphQLError: Must be at least 0],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`FIELD_DEFINITION Int validate @constraint #min0 should pass 1`] = `
Object {
  "data": Object {
    "books": Array [
      Object {
        "title": 1,
      },
      Object {
        "title": 2,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`FIELD_DEFINITION Int validate @constraint #multipleOf should fail 1`] = `
Object {
  "data": Object {
    "books": Array [
      Object {
        "title": null,
      },
      Object {
        "title": 2,
      },
      Object {
        "title": null,
      },
    ],
  },
  "errors": Array [
    [GraphQLError: Must be a multiple of 2],
    [GraphQLError: Must be a multiple of 2],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`FIELD_DEFINITION Int validate @constraint #multipleOf should pass 1`] = `
Object {
  "data": Object {
    "books": Array [
      Object {
        "title": 2,
      },
      Object {
        "title": 4,
      },
      Object {
        "title": 6,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`INPUT_FIELD_DEFINITION Int validate @constraint #exclusiveMax should fail 1`] = `
Object {
  "data": undefined,
  "errors": Array [
    [GraphQLError: Variable "$input" got invalid value 3 at "input.title"; Expected type ConstraintNumber. Must be less than 3],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`INPUT_FIELD_DEFINITION Int validate @constraint #exclusiveMax should pass 1`] = `
Object {
  "data": Object {
    "createBook": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`INPUT_FIELD_DEFINITION Int validate @constraint #exclusiveMin should fail 1`] = `
Object {
  "data": undefined,
  "errors": Array [
    [GraphQLError: Variable "$input" got invalid value 3 at "input.title"; Expected type ConstraintNumber. Must be greater than 3],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`INPUT_FIELD_DEFINITION Int validate @constraint #exclusiveMin should pass 1`] = `
Object {
  "data": Object {
    "createBook": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`INPUT_FIELD_DEFINITION Int validate @constraint #max should fail 1`] = `
Object {
  "data": undefined,
  "errors": Array [
    [GraphQLError: Variable "$input" got invalid value 5 at "input.title"; Expected type ConstraintNumber. Must be no greater than 3],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`INPUT_FIELD_DEFINITION Int validate @constraint #max should pass 1`] = `
Object {
  "data": Object {
    "createBook": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`INPUT_FIELD_DEFINITION Int validate @constraint #min should fail 1`] = `
Object {
  "data": undefined,
  "errors": Array [
    [GraphQLError: Variable "$input" got invalid value 2 at "input.title"; Expected type ConstraintNumber. Must be at least 3],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`INPUT_FIELD_DEFINITION Int validate @constraint #min should pass 1`] = `
Object {
  "data": Object {
    "createBook": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`INPUT_FIELD_DEFINITION Int validate @constraint #multipleOf should fail 1`] = `
Object {
  "data": undefined,
  "errors": Array [
    [GraphQLError: Variable "$input" got invalid value 7 at "input.title"; Expected type ConstraintNumber. Must be a multiple of 2],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`INPUT_FIELD_DEFINITION Int validate @constraint #multipleOf should pass 1`] = `
Object {
  "data": Object {
    "createBook": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
